src/webRTCEndpoint.ts

```diff
@@ -421,20 +425,25 @@ export class WebRTCEndpoint extends (EventEmitter as new () => TypedEmitter<
       case "tracksAdded":
         data = deserializedMediaEvent.data;
         if (this.getEndpointId() === data.endpointId) return;
-        data.trackIdToMetadata = new Map<string, any>(
-          Object.entries(data.trackIdToMetadata)
-        );
+        data.tracks = new Map<string, any>(
+          Object.entries(data.tracks)
+        )
         endpoint = this.idToEndpoint.get(data.endpointId)!;
-        const oldTrackIdToMetadata = endpoint.trackIdToMetadata;
-        endpoint.trackIdToMetadata = new Map([
-          ...endpoint.trackIdToMetadata,
-          ...data.trackIdToMetadata,
+        const oldTracks = endpoint.tracks;
+
+        data.tracks = this.mapMediaEventTracksToTrackContextImpl(data.tracks, endpoint)
+
+        endpoint.tracks = new Map([
+          ...endpoint.tracks,
+          ...data.tracks,
         ]);
```

Comment:
I feel like this code could be somehow refactored. Some ideas that come to my mind:
* move ME parsing into a separate module/file
* add TrackContextImpl method e.g. `fromMediaEvent`
* make `mapMediaEventTracksToTrackContextImpl` shorter